### PR TITLE
debug fixes

### PR DESCRIPTION
**Pull request by dsyme**
I noticed a few problems while running VisualFSharp tools in debug mode on some F# scripts.  These fixes don't affect the actual product behaviour

1. We are getting an assert in the MSBuild resolution while processing scripts.  The problem is that old broken code to compute the target framework version is kicking in and evaluating to `"v4.0"` which is no longer even supported.  This computation of target framework version is only relevant in secondary stages of processing scripts and should just be HighestInstalledFrameworkVersion

2. The Simulated MSBuild code was failing if the gac directory does not exist.  This code is only used by FCS. and was only failing for the .NET Standard/Core version of FCS  This is now protected bya `Directory.Exists`

### CODE REVIEW

File: src/fsharp/SimulatedMSBuildReferenceResolver.fs [hunk 1]
```diff
@@ -147,35 +147,37 @@ let internal SimulatedMSBuildResolver =
...
-                                [ for gacdir in Directory.EnumerateDirectories(gac) do 
-                                    let assdir = Path.Combine(gacdir,n.Name)
-                                    if Directory.Exists(assdir) then 
-                                        for tdir in Directory.EnumerateDirectories(assdir) do 
-                                            let trialPath = Path.Combine(tdir,qual)
-                                            if FileSystem.SafeExists(trialPath) then 
-                                                yield trialPath ]
+                                [ if Directory.Exists(gac) then 
+                                    for gacdir in Directory.EnumerateDirectories(gac) do 
+                                        let assdir = Path.Combine(gacdir,n.Name)
```
Comment (cartermp) [inline src/fsharp/SimulatedMSBuildReferenceResolver.fs]:
```suggestion
                                        let assemblyDir = Path.Combine(gacdir,n.Name)
```

:)

File: src/fsharp/SimulatedMSBuildReferenceResolver.fs [hunk 2]
```diff
@@ -147,35 +147,37 @@ let internal SimulatedMSBuildResolver =
...
-
-                                    if Directory.Exists(verdir) then 
-                                        let trialPath = Path.Combine(verdir,qual)
-                                        //printfn "searching GAC: %s" trialPath
-                                        if FileSystem.SafeExists(trialPath) then 
-                                            success trialPath
+                            if Directory.Exists(gac) then 
+                                for gacdir in Directory.EnumerateDirectories(gac) do 
+                                    //printfn "searching GAC directory: %s" gacdir
+                                    let assdir = Path.Combine(gacdir,n.Name)
```
Comment (cartermp) [inline src/fsharp/SimulatedMSBuildReferenceResolver.fs]:
```suggestion
                                    let assembnlyDir = Path.Combine(gacdir,n.Name)
```

:)

File: src/fsharp/SimulatedMSBuildReferenceResolver.fs [hunk 3]
```diff
@@ -147,35 +147,37 @@ let internal SimulatedMSBuildResolver =
...
+                                    if Directory.Exists(assdir) then
```
Comment (cartermp) [inline src/fsharp/SimulatedMSBuildReferenceResolver.fs]:
```suggestion
                                    if Directory.Exists(assemblyDir) then 
```

:)

File: src/fsharp/SimulatedMSBuildReferenceResolver.fs [hunk 4]
```diff
@@ -147,35 +147,37 @@ let internal SimulatedMSBuildResolver =
...
-                                    if Directory.Exists(assdir) then 
-                                        for tdir in Directory.EnumerateDirectories(assdir) do 
-                                            let trialPath = Path.Combine(tdir,qual)
-                                            if FileSystem.SafeExists(trialPath) then 
-                                                yield trialPath ]
+                                [ if Directory.Exists(gac) then 
+                                    for gacdir in Directory.EnumerateDirectories(gac) do 
+                                        let assdir = Path.Combine(gacdir,n.Name)
+                                        if Directory.Exists(assdir) then 
+                                            for tdir in Directory.EnumerateDirectories(assdir) do
```
Comment (cartermp) [inline src/fsharp/SimulatedMSBuildReferenceResolver.fs]:
```suggestion
                                            for tdir in Directory.EnumerateDirectories(assemblyDir) do 
```

:)

File: src/fsharp/SimulatedMSBuildReferenceResolver.fs [hunk 5]
```diff
@@ -147,35 +147,37 @@ let internal SimulatedMSBuildResolver =
...
-                                    let assdir = Path.Combine(gacdir,n.Name)
-                                    if Directory.Exists(assdir) then 
-                                        for tdir in Directory.EnumerateDirectories(assdir) do 
-                                            let trialPath = Path.Combine(tdir,qual)
-                                            if FileSystem.SafeExists(trialPath) then 
-                                                yield trialPath ]
+                                [ if Directory.Exists(gac) then 
+                                    for gacdir in Directory.EnumerateDirectories(gac) do 
+                                        let assdir = Path.Combine(gacdir,n.Name)
+                                        if Directory.Exists(assdir) then
```
Comment (cartermp) [inline src/fsharp/SimulatedMSBuildReferenceResolver.fs]:
```suggestion
                                        if Directory.Exists(assemblyDir) then 
```
:)

**Comment by KevinRansom** [comment]
@dsyme 

````
        /home/vsts/work/1/s/src/fsharp/CompileOps.fs(2792,9): error FS1182: The value 'ilGlobals' is unused [/home/vsts/work/1/s/fcs/FSharp.Compiler.Service/FSharp.Compiler.Service.fsproj]
2019-01-31T13:48:56.1556589Z 
````

**Pull request merged**